Automattic/wp-calypso
# Live Preview: "Install&Preview" themes on Atomic sites

Close #80089
Related to https://github.com/Automattic/wp-calypso/pull/80163, https://github.com/WordPress/gutenberg/issues/50749

## Proposed Changes

This PR allows Atomic sites to preview non-installed themes by installing the previewing theme on demand. 

- This PR is only for Atomic sites and NOT for Simple sites. We can address it for Simple sites in the later iteration. (c.f. https://github.com/Automattic/wp-calypso/issues/80411)
- This PR is NOT for externally managed themes (= partner themes), as we should not install such a theme if a user is not subscribed to the theme. We need to think about another solution for externally managed themes.

## Testing Instructions



- Create an Atomic site
- Go to the Live Preview by clicking the "Live Preview" on the Theme Detail page (or clicking the theme option on the Theme Showcase page)
- Verify that you can preview non-installed themes, such as;
	- the Site Editor compatible free non-installed themes
	- the Site Editor compatible premium/woo non-installed themes (e.g. Yuga, Tazza)
	- the Site Editor compatible wporg non-installed themes (e.g. Crossfit Block)
	- the Site Editor compatible Partner INSTALLED themes (e.g. Bonum, Nokul, Macchiato)
		- You should not see the "Live Preview" for non-installed partner themes

https://github.com/Automattic/wp-calypso/assets/5287479/8892bac6-a5b0-4753-8d0c-24fd54e0fed5

https://github.com/Automattic/wp-calypso/assets/5287479/0c37131c-c716-4e21-a2cb-bebce9a0b916

## Pre-merge Checklist
